tests/std/tests/P1899R3_views_stride/test.cpp

```diff
@@ -0,0 +1,609 @@
+            [[maybe_unused]] const iterator_t<const R> const_defaulted;
+        }
+        auto i                               = r.begin();
+        same_as<iterator_t<const R>> auto ci = as_const(r).begin();
+
+        // Comparisons with std::default_sentinel_t
+        assert(ci != default_sentinel);
+        if constexpr (forward_range<const R>) {
+            assert(ranges::next(ci, r.end()) == default_sentinel);
+        }
+
+        // Equality operators
+        if constexpr (equality_comparable<iterator_t<const V>>) {
+            assert(ci == ci);
+            if constexpr (forward_range<const R>) {
+                assert(ci != ranges::next(ci, 1));
+            }
+
+            if constexpr (constructible_from_nonconst) {
+                assert(ci == i);
+                if constexpr (forward_range<const R>) {
+                    assert(ci != ranges::next(i, 1));
+                }
+            }
+        }
+
+        if constexpr (forward_range<const R>) {
+            assert(*ci++ == expected[0]);
+        } else {
+            STATIC_ASSERT(is_void_v<decltype(i++)>);
+        }
+        assert(*++ci == expected[2]);
+
+        if constexpr (bidirectional_range<const R>) {
+            assert(*ci-- == expected[2]);
+            assert(*--ci == expected[0]);
+        }
+
+        if constexpr (random_access_range<const R>) {
+            ci += 2;
+            assert(*ci == expected[2]);
+
+            ci -= 2;
+            assert(*ci == expected[0]);
+
+            assert(ci[2] == expected[2]);
+
+            const auto ci2 = ci + 2;
+            assert(*ci2 == expected[2]);
+
+            const auto ci3 = 2 + ci;
+            assert(*ci3 == expected[2]);
+
+            const auto ci4 = ci3 - 2;
+            assert(*ci4 == expected[0]);
+
+            const same_as<ranges::range_difference_t<V>> auto diff1 = ci2 - ci;
+            assert(diff1 == 2);
+
+            const same_as<ranges::range_difference_t<V>> auto diff2 = ci - ci2;
+            assert(diff2 == -2);
+
+            // Comparisons
+            assert(ci < ci2);
+            assert(ci <= ci2);
+            assert(ci2 > ci);
+            assert(ci2 >= ci);
+            if constexpr (three_way_comparable<iterator_t<const V>>) {
+                assert(ci <=> ci4 == strong_ordering::equal);
+                assert(ci <=> ci2 == strong_ordering::less);
+                assert(ci2 <=> ci == strong_ordering::greater);
+            }
+
+            // Cross comparisons
+            if constexpr (constructible_from_nonconst) {
+                assert(i < ci2);
+                assert(i <= ci2);
+                assert(ci2 > i);
+                assert(ci2 >= i);
+                if constexpr (three_way_comparable<iterator_t<const V>>) {
+                    assert(ci <=> ci2 == strong_ordering::less);
+                    assert(ci <=> ci4 == strong_ordering::equal);
+                    assert(ci2 <=> ci == strong_ordering::greater);
+                }

```

Comment:
It appears that the spaceship tests here aren't testing cross-comparisons between const and nonconst. Was it intended to use `i` instead of `ci` here?

Reply:
> Was it intended to use `i` instead of `ci` here?

Yes, my bad